web/autolinker.js

```diff
@@ -96,21 +96,32 @@ class Autolinker {
   static #regex;
 
   static findLinks(text) {
-    // Regex can be tested and verified at https://regex101.com/r/zgDwPE/1.
+    // Regex can be tested and verified at https://regex101.com/r/rXoLiT/2.
     this.#regex ??=
-      /\b(?:https?:\/\/|mailto:|www\.)(?:[[\S--\[]--\p{P}]|\/|[\p{P}--\[]+[[\S--\[]--\p{P}])+|\b[[\S--@]--\{]+@[\S--.]+\.[[\S--\[]--\p{P}]{2,}/gmv;
+      /\b(?:https?:\/\/|mailto:|www\.)(?:[\S--[\p{P}<>]]|\/|[\S--[\[\]]]+[\S--[\p{P}<>]])+|\b[\S--[@\p{Ps}\p{Pe}<>]]+@([\S--[\p{P}<>]]+(?:\.[\S--[\p{P}<>]]+)+)/gmv;
 
     const [normalizedText, diffs] = normalize(text);
     const matches = normalizedText.matchAll(this.#regex);
     const links = [];
     for (const match of matches) {
-      const raw =
+      let raw;
+      if (
         match[0].startsWith("www.") ||
-        match[0].startsWith("mailto:") ||
         match[0].startsWith("http://") ||
         match[0].startsWith("https://")
-          ? match[0]
-          : `mailto:${match[0]}`;
+      ) {
+        raw = match[0];
+      } else {
+        // Check if the email address is valid through the URL constructor.
+        // In the future, this case be replaced by `URL.canParse`.
```

Comment:
It seems that core-js has support for `URL.canParse` already, see https://github.com/zloirock/core-js?tab=readme-ov-file#url-and-urlsearchparams, so we might be able to use it directly?